ballerina/utils.bal

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/persist;
+import ballerina/jballerina.java;
+import ballerinax/googleapis.sheets;
+
+
+isolated function getKeyFromAlreadyExistsErrorMessage(string errorMessage) returns string|persist:Error {
+    int? startIndex = errorMessage.indexOf(".Duplicate entry '");
+    int? endIndex = errorMessage.indexOf("' for key");
+
+    if startIndex is () || endIndex is () {
+        return <persist:Error>error("Unable to determine key from DuplicateKey error message.");
+    }
+
+    string key = errorMessage.substring(startIndex + 18, endIndex);
+    return key;
+}
+
+isolated function convertToArray(typedesc<record {}> elementType, record {}[] arr) returns elementType[] = @java:Method {
+    'class: "io.ballerina.stdlib.persist.googlesheets.Utils"
+} external;
```

Comment:
Can we use the same util method in the persist module? To avoid duplicate code

Reply:
Updated in [4154bd8](https://github.com/ballerina-platform/module-ballerinax-persist.googlesheets/pull/1/commits/4154bd8d3c4e336f435e1143a2da09e7129c858c)